src/shaders/globe_depth.fragment.glsl

```diff
@@ -0,0 +1,20 @@
+#ifdef GL_ES
+precision highp float;
+#endif
+
+// Pack depth to RGBA. A piece of code copied in various libraries and WebGL
+// shadow mapping examples.
+vec4 pack_depth(float ndc_z) {
+    float depth = ndc_z * 0.5 + 0.5;
+    const vec4 bit_shift = vec4(256.0 * 256.0 * 256.0, 256.0 * 256.0, 256.0, 1.0);
+    const vec4 bit_mask  = vec4(0.0, 1.0 / 256.0, 1.0 / 256.0, 1.0 / 256.0);
+    vec4 res = fract(depth * bit_shift);
+    res -= res.xxyz * bit_mask;
+    return res;
+}
```

Comment:
This function could be shared as it's the same function used in [terrain_depth.fragment](https://github.com/mapbox/mapbox-gl-js/blob/e8c67cddb023bb860b37ae00330a866c40f20830/src/shaders/terrain_depth.fragment.glsl#L7), maybe a good opportunity to put that in the shared shader prelude to reduce duplication.